ArithmeticTools/BitwiseOperations.swift

```diff
@@ -19,10 +19,56 @@ public func log2(powerOfTwo x: Int) -> Int {
 /// This retrieves the right-hand-side value, then inverts this value in relation to the given
 /// `max` value. The returned value can then be re-shifted as necessary.
 ///
-/// ```
-/// invert(powerOfTwo: 1 << 3, within: 4) // 1
-/// ```
+///     invert(powerOfTwo: 1 << 3, within: 4) // 1
+///
 public func invert(powerOfTwo: Int, within max: Int) -> Int {
     let unrolled = log2(powerOfTwo: powerOfTwo)
     return max - unrolled
 }
+
+private let intBitCount = MemoryLayout<Int>.size * 8
+
+/// Count Trailing Zeros (ctz) counts the number of zero bits succeeding the least
+/// significant one bit. For example, the ctz of 0x00000F00 is 8, and the ctz of
+/// 0x80000000 is 31. This also counts the exponent of the `2` factor in the prime
+/// factorization of a positive number.
+public func countTrailingZeros(_ n: Int) -> Int {
+
+	var mask = 1
+
+	for index in 0...intBitCount {
+		
+        if mask & n != 0 {
+			return index
+		}
+        
+		mask <<= 1
+	}
+
+	return intBitCount
+}
+
+/// Count Leading Zeros (clz) counts the number of zero bits preceding the most
+/// significant one bit. For example, the clz of 0x00F00000 is 8, and the clz of
+/// 0x00000001 is 31.
+public func countLeadingZeros(_ n: Int) -> Int {
+    
+	var mask = 1 << (intBitCount - 1)
+
+	for index in 0...intBitCount {
+		
+        if mask & n != 0 {
+			return index
+		}
+        
+        mask >>= 1
```

Comment:
this needs to use logical right shift (so either `mask = mask >>> 1` or `mask >>>>= 1` if you want to re-implement that operator).